include/biosHandler.hpp

```diff
@@ -4,16 +4,81 @@
 
 namespace vpd
 {
+
+/**
+ * @brief Interface class for BIOS handling.
+ *
+ * The class layouts virtual methods required to be implemented by any concrete
+ * class intended to use feature provided via BIOS handler class.
+ */
+class BiosHandlerInterface
+{
+  public:
+    /**
+     * @brief API to back up or restore BIOS attributes.
+     *
+     * Concrete class should implement the API and read the backed up data from
+     * its designated location and take a call if it should be backed up or
+     * restored.
+     */
+    virtual void backUpOrRestoreBiosAttributes() = 0;
+
+    /**
+     * @brief API to be triggered on BIOS attribute change.
+     *
+     * Concrete class should implement the API to capture any change in the BIOS
+     * attribues and in the implementation needs to filter out the attribute it
+     * is interested.
+     *
+     * @param[in] i_msg - The callback message.
+     */
+    virtual void biosAttribsCallback(sdbusplus::message_t& i_msg) = 0;
+};
+
+/**
```

Comment:
a) Then we again end up with company specific file design which we don't want.
b) These are small classes, will just have function declarations.
c) Also these classes are interrelated so keeping it in a single file should help understand the dependency.
Let me know your thoughts.

Reply:
my question was for the below class IbmBiosHandler only since each company has to add their own one ,  the file will have too many companies specific changes

Reply:
Agree but for now we don't know how many are going to enroll for this feature. Also was trying to avoid company specific file names.
In future if we see many changes coming in here, we will split it. For now changes are too small to create a new company specific file.